packages/core/admin/admin/src/content-manager/components/DynamicTable/CellContent/Relation/index.js

```diff
@@ -1,88 +1,128 @@
 
 const SINGLE_RELATIONS = ['oneToOne', 'manyToOne'];
 
-const ActionWrapper = styled.span`
-  svg {
-    height: ${4 / 16}rem;
-  }
+const TypographyMaxWidth = styled(Typography)`
+  max-width: 500px;
 `;
 
-const RelationCountBadge = styled(Badge)`
-  display: flex;
-  align-items: center;
-  height: ${20 / 16}rem;
-  width: ${16 / 16}rem;
-`;
+const fetchRelation = async (endPoint, notifyStatus) => {
+  const {
+    data: { results, pagination },
+  } = await axiosInstance.get(endPoint);
+
+  notifyStatus();
+
+  return { results, pagination };
+};
 
 const Relation = ({ fieldSchema, metadatas, queryInfos, name, rowId, value }) => {
   const { formatMessage } = useIntl();
-  const [visible, setVisible] = useState(false);
-  const buttonRef = useRef();
+  const { notifyStatus } = useNotifyAT();
+  const requestURL = getRequestUrl(`${queryInfos.endPoint}/${rowId}/${name.split('.')[0]}`);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const Label = (
+    <>
+      <Badge>{value.count}</Badge>{' '}
+      {formatMessage(
+        {
+          id: 'content-manager.containers.ListPage.items',
+          defaultMessage: '{number, plural, =0 {items} one {item} other {items}}',
+        },
+        { number: value.count }
+      )}
+    </>
+  );
+
+  const notify = () => {
+    const message = formatMessage({
+      id: getTrad('DynamicTable.relation-loaded'),
+      defaultMessage: 'The relations have been loaded',
+    });
+    notifyStatus(message);
+  };
+
+  const { data, status } = useQuery(
+    [fieldSchema.targetModel, rowId],
+    () => fetchRelation(requestURL, notify),
+    {
+      staleTime: 0,
+    },
+    {
+      enabled: isOpen,
+    }
+  );
 
   if (SINGLE_RELATIONS.includes(fieldSchema.relation)) {
     return (
       <Typography textColor="neutral800">
-        <CellValue type={metadatas.mainField.schema.type} value={value[metadatas.mainField.name]} />
+        <CellValue
+          type={metadatas.mainField.schema.type}
+          value={value[metadatas.mainField.name] || value.id}
+        />
       </Typography>
     );
   }
 
-  const handleTogglePopover = () => setVisible(prev => !prev);
-
   return (
-    <Flex {...stopPropagation}>
-      <RelationCountBadge>{value.count}</RelationCountBadge>
-      <Box paddingLeft={2}>
-        <Typography textColor="neutral800">
-          {formatMessage(
-            {
-              id: 'content-manager.containers.ListPage.items',
-              defaultMessage: '{number, plural, =0 {items} one {item} other {items}}',
-            },
-            { number: value.count }
-          )}
-        </Typography>
-      </Box>
-      {value.count > 0 && (
-        <ActionWrapper>
-          <IconButton
-            onClick={handleTogglePopover}
-            ref={buttonRef}
-            noBorder
-            label={formatMessage({
-              id: 'content-manager.popover.display-relations.label',
-              defaultMessage: 'Display relations',
+    <Box {...stopPropagation}>
```

Comment:
What do you think about splitting this component into 2 different ones
- Single
- Multiple

in order to improve readability and to make functional testing easier?

I think that the rendering of a single Relation is way simpler than a multiple so it makes sense for me to create a specific for it.

Reply:
Also do you think it could be worth to add some tests?

Reply:
I'm not sure what you mean with your first comment. I've copied all the functional code over and didn't change it at all and I don't see any difference in the component rendering. There is no branching based on single- or multiple relations happening.

As for your second comment: added a basic rendering + data fetching test in eb8d57a37baaff5ea3ee183f52c25d860f4e5e37

Reply:
Both are very good suggestions. 👍🏼 Added them in subsequent commits.